database_manager/src/run_server.cpp

```diff
@@ -775,14 +776,17 @@ bool remove_facts_to_agent_db(std::string agentId, std::vector<toaster_msgs::Fac
             }
             //and add a new event
             if(agentId == mainAgent){
-               sql = (std::string)"INSERT INTO events_table (subject_id,predicate,propertyType,target_id,observability,confidence,time) VALUES ('"
-                       + boost::lexical_cast<std::string>(it->subjectId) + "','!"
-                       + (std::string)it->property + "','"
-                       + (std::string)it->propertyType + "','"
-                       + boost::lexical_cast<std::string>(it->targetId) + "',"
-                       + boost::lexical_cast<std::string>(it->factObservability) + ","
-                       + boost::lexical_cast<std::string>(it->confidence) + ","
-                       + boost::lexical_cast<std::string>(it->time) + ")";
+            timeval curTime;
+            gettimeofday(&curTime, NULL);
+            unsigned long now = curTime.tv_sec * pow(10,9) + curTime.tv_usec;
```

Comment:
Please make this last:
facts is ros::time::now() (e.g. https://github.com/Greg8978/toaster/blob/master/pdg/src/Pr2RobotReader.cpp l.89)